services-core/src/economic_viability.rs

```diff
@@ -229,11 +238,67 @@ impl EconomicViabilityComputing for PriorityEconomicViabilityComputer {
     }
 }
 
+/// Use the better result of two computers. If first fails second is used.
+/// For min-avg-fee lower is better, for max-gas-price higher is better.
+pub struct BoundedEconomicViabilityComputer {
+    first: Box<dyn EconomicViabilityComputing>,
+    second: Box<dyn EconomicViabilityComputing>,
+}
+
+impl BoundedEconomicViabilityComputer {
+    pub fn new(
+        first: Box<dyn EconomicViabilityComputing>,
+        second: Box<dyn EconomicViabilityComputing>,
+    ) -> Self {
+        Self { first, second }
+    }
+
+    async fn bounded<'a, T, Future>(
+        &'a self,
+        operation: impl Fn(&'a dyn EconomicViabilityComputing) -> Future + 'a,
+        comparison: impl Fn(T, T) -> T,
+    ) -> Result<T>
+    where
+        Future: std::future::Future<Output = Result<T>>,
+    {
+        let (first, second) = future::join(
+            operation(self.first.as_ref()),
+            operation(self.second.as_ref()),
+        )
+        .await;
+        let second = second?;
```

Comment:
I find it quite arbitrary that we fall back to the second one if the first fails but fail if the second passes. Maybe making it clear that one is the `bound` (which we fall back to) and the other one is the `optimistic` (I'm lacking a good name here) would make that seem less arbitrary?